src/server/svr_jobfunc.c

```diff
@@ -5964,6 +5964,86 @@ send_job_exec_update_to_mom(job *pjob, char *err_msg, int err_msg_sz,
 	return (rc);
 }
 
+/**
+ *
+ * @brief
+ *	Extracts mom hostnames from exechostx and adds it to list
+ *
+ * @param[in/out]	to_head - destination reliable_job_node list
+ * @param[in]	exechostx - string in exechost format
+ *
+ * @return 	void
+ */
+static void
+populate_mom_list(pbs_list_head *to_head, char *exechostx)
+{
+	char *hostn = NULL, *last = NULL, *peh;
+	int  hasprn = 0;
+
+	if ((to_head == NULL) || (exechostx == NULL) || !*exechostx)
+		return;
+
+	peh = strdup(exechostx);
+	if (peh == NULL) {
+		log_err(errno, __func__, "strdup error");
+		return;
+	}
+
+	CLEAR_HEAD((*to_head));
+
+	for (hostn = parse_plus_spec_r(peh, &last, &hasprn);
+		hostn;
+		hostn = parse_plus_spec_r(last, &last, &hasprn) ) {
+		if (reliable_job_node_add(to_head, strtok(hostn, ":/")) == -1) {
+			free(peh);
+			return;
+		}
+	}
+	free(peh);
+	return;
+}
+
+/**
+ * @brief
+ *	returns a copy of partial select string representing the MS (first) chunk
+ *	Note - caller to free the returned string pointer
+ *
+ * @param[in]		select_str - pointer to complete schedselect string
+ *
+ * @return char *
+ * @retval ptr	pointer to malloc'd string containing ms (first) chunk's select str
+ *
+ * @note
+ * caller to free the returned string pointer
+*/
+static char *
+get_ms_select_chunk(char *select_str)
+{
+	char *slast, *selbuf, *psubspec, *retval = NULL;
+	int   hpn;
+
+    if (select_str == NULL) {
+    	log_err(-1, __func__, "bad param passed");
+    	return (NULL);
```

Comment:
Indentation looks off here and in other parts of this function.

Reply:
Thanks for the sharp eye. Fixed in commit https://github.com/PBSPro/pbspro/pull/1452/commits/8f309272ad00f7587c830cad07238cb40598e24a